agent/exploits/cve_2023_28461.py

```diff
@@ -0,0 +1,107 @@
+"""Agent Asteroid implementation for CVE-2023-28461"""
+
+import datetime
+import logging
+import re
+import ssl
+from typing import Any
+from urllib import parse as urlparse
+
+import urllib3
+from packaging import version
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+VULNERABILITY_TITLE = (
+    "Array Networks Array AG Series and vxAG Remote Code Execution Vulnerability"
+)
+VULNERABILITY_REFERENCE = "CVE-2023-28461"
+VULNERABILITY_DESCRIPTION = """A critical vulnerability in Array Networks Array AG Series and vxAG
+SSL VPN gateways allows remote code execution by exploiting an HTTP header with the 'flags' attribute
+to browse the filesystem without authentication."""
+RISK_RATING = "CRITICAL"
+MAX_VULNERABLE_VERSION = version.parse("9.4.0.481")
+
+DEFAULT_TIMEOUT = datetime.timedelta(seconds=90)
+
+
+@exploits_registry.register
+class CVE202328461Exploit(webexploit.WebExploit):
+    accept_request = definitions.Request(
+        method="GET", path="/prx/000/http/localhost/login"
+    )
+    accept_pattern = [re.compile(r"Rel_AG_(\d+_\d+_\d+_\d+)")]
+    metadata = definitions.VulnerabilityMetadata(
+        title=VULNERABILITY_TITLE,
+        description=VULNERABILITY_DESCRIPTION,
+        reference=VULNERABILITY_REFERENCE,
+        risk_rating=RISK_RATING,
+    )
+
+    def _create_ssl_context(self) -> Any:
+        """Creates an SSL context that allows insecure connections."""
+        ctx = urllib3.util.ssl_.create_urllib3_context()
+        ctx.load_default_certs()
+        ctx.options |= 0x4  # ssl.OP_LEGACY_SERVER_CONNECT
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
+        return ctx
+
+    def _send_ssl_request(self, url: str) -> Any:
+        """Send a GET request using urllib3 with a custom SSL context."""
+        ssl_context = self._create_ssl_context()
+        with urllib3.PoolManager(ssl_context=ssl_context) as http:
+            try:
+                response = http.request(
+                    "GET", url, timeout=DEFAULT_TIMEOUT.total_seconds()
+                )
+                return response.data.decode("utf-8")
+            except urllib3.exceptions.HTTPError as e:
+                logging.error("Error during SSL request: %s", e)
+                return ""
```

Comment:
Both methods are not using self and are not related to the class take them outside.